mmselfsup/models/utils/knn_classifier.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+
+# This file is borrowed from
+# https://github.com/facebookresearch/dino/blob/main/eval_knn.py
+
+import torch
+import torch.nn as nn
+
+
+@torch.no_grad()
+def knn_classifier(train_features,
+                   train_labels,
+                   test_features,
+                   test_labels,
+                   k,
+                   T,
+                   num_classes=1000):
+    top1, top5, total = 0.0, 0.0, 0
+    train_features = nn.functional.normalize(train_features, dim=1)
+    test_features = nn.functional.normalize(test_features, dim=1)
+    train_features = train_features.t()
+    num_test_images, num_chunks = test_labels.shape[0], 100
+    imgs_per_chunk = num_test_images // num_chunks
```

Comment:
Maybe you want to prevent the memory overflow problem by splitting all images into several chunks. Better to add some comments here to be more self-contained.